src/ngraph/ngraph_emitter.cc

```diff
@@ -695,6 +695,42 @@ void Emitter::CreateBinaryOps() {
     return cast_result(CreateAutoBroadcast<ngraph::op::LessEq>(node),
                        getType(node->dtype_));
   };
+  ngraph_op_funcs_["smooth_l1"] = [this](const NodePtr& node) {
+    /* Smooth L1 Loss is a loss specific for R-CNN franchise training
+     * Smooth L1 Loss function:
+     * f(x) = 0.5 * (sigma * x) ^ 2,     |x| < 1 / sigma^2
+     *      = |x| - 0.5 / sigma / sigma, otherwise
+     * When sigma = 1, it is equivalent to the Huber loss, evaluated at
+     * delta = 1.
+     * smooth_l1_loss = w_out * f(w_in * x)
+     * with w_in, w_out provided by input_data.
+     */
+    auto input = op_map_[node->inputs_[0]];
+    auto sigma =
+        makeConstant(node, get_default(node, "scalar", std::string("0")));
+    auto sigma_sq = sigma * sigma;
+    auto inv_sigma_sq = std::make_shared<ngraph::op::Divide>(
+        makeConstant(node, "1.0"), sigma_sq);
```

Comment:
Can/should we sniff for the value of sigma being zero?

We *can* do that sniffing here because it's a graph-compile-time constant.

We *maybe* *should* do that sniffing here because it's often a good idea to let users know about problems ASAP, so they can better identify  what they're doing wrong on their end.

Reply:
formula does not treat zero as special for sigma, it is same as any other `scalar` value the user chooses to pass. we should not restrict users from using this formula with zero (as formula or op description does not say so)

Reply:
👍 FYI, I created #253 for follow-up.